java/algorithm/src/test/java/org/apache/arrow/algorithm/sort/TestStableVectorComparator.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.algorithm.sort;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.VarCharVector;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Test cases for {@link StableVectorComparator}.
+ */
+public class TestStableVectorComparator {
+
+  private BufferAllocator allocator;
+
+  @Before
+  public void prepare() {
+    allocator = new RootAllocator(1024 * 1024);
+  }
+
+  @After
+  public void shutdown() {
+    allocator.close();
+  }
+
+  @Test
+  public void testCompare() {
+    try (VarCharVector vec = new VarCharVector("", allocator)) {
+      vec.allocateNew(100, 5);
+      vec.setValueCount(10);
+
+      // fill data to sort
+      vec.set(0, "ba".getBytes());
+      vec.set(1, "abc".getBytes());
+      vec.set(2, "aa".getBytes());
+      vec.set(3, "abc".getBytes());
+      vec.set(4, "a".getBytes());
+
+      VectorValueComparator<VarCharVector> comparator = new TestVarCharSorter();
+      VectorValueComparator<VarCharVector> stableComparator = new StableVectorComparator<>(comparator);
+      stableComparator.attachVector(vec);
+
+      assertTrue(stableComparator.compare(0, 1) > 0);
+      assertTrue(stableComparator.compare(1, 2) < 0);
+      assertTrue(stableComparator.compare(2, 3) < 0);
+      assertTrue(stableComparator.compare(1, 3) < 0);
+      assertTrue(stableComparator.compare(3, 1) > 0);
```

Comment:
can you add a test for equality (same index?)

Reply:
Sure. Thanks.